### PR TITLE
Tfm update for new build system

### PR DESCRIPTION
**Pull request by ioannisg**
Required fixes in Zephyr in the wake of upgrading TF-M to the new build system (+ getting support for nRF devices).

The upmerge is present in: https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/15

The following platforms can currently build and run the TFM Integration samples successfully:
- mps2_an521_nonsecure
- nrf5340pdk_nrf5340_cpuappns
- nrf9160dk_nrf9160ns
- nucleo_l552ze_q_ns

The following platforms now also build successfully (but I have no HW to test)
- lpcxpresso55s69
- musca_b1

The PR also contains the required doc changes for samples and boards.

### CODE REVIEW

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt [hunk 1]
```diff
@@ -27,36 +28,36 @@ if (CONFIG_BUILD_WITH_TFM)
...
 			 --align 1
 			 -v ${TFM_IMAGE_VERSION_S}
+			 --pad
+			 --pad-header
 			 ${ADD_NS_IMAGE_MIN_VER}
-			 ${ADD_SECURITY_COUNTER_S}
+			 -s auto
 			 -H 0x400
-			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/STM_NUCLEO_L552ZE_Q/tfm_s.bin
+			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/MPS2/AN521/tfm_s.bin
```
Comment (erwango) [inline boards/arm/nucleo_l552ze_q/CMakeLists.txt:42]: I guess this change is not correct, but I'm not sure if it should be `stm/nucleo_l552ze_q` of  the existing line was fine

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt (hunk 1 above)
Comment (agansari) [reply boards/arm/nucleo_l552ze_q/CMakeLists.txt:42]: Looks like we don't actually need custom vendor/board paths, binaries are copied to ./build/tfm/bin/
This could be used further to unify the merge script.

Comment (ioannisg) [reply boards/arm/nucleo_l552ze_q/CMakeLists.txt:42]: That's probably true @agansari, we can revisit this as part of the cleanup (something along what @anangl had done in https://github.com/zephyrproject-rtos/zephyr/pull/28806)

I'd just fix this for now, since I want to separate the upmerge from the cleanup 👍

Comment (ioannisg) [reply boards/arm/nucleo_l552ze_q/CMakeLists.txt]: This is fixed, now, @erwango

File: boards/arm/mps2_an521/CMakeLists.txt [hunk 2]
```diff
@@ -18,7 +18,8 @@ if (CONFIG_BUILD_WITH_TFM)
 		set(TFM_IMAGE_VERSION_NS 0.0.0+0)
 	endif()
 
-	set(PREPROCESSED_FILE "${CMAKE_BINARY_DIR}/tfm/image_macros_preprocessed")
+	set(PREPROCESSED_FILE_S "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o")
+	set(PREPROCESSED_FILE_NS "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o")
```
Comment (tejlmand) [inline boards/arm/mps2_an521/CMakeLists.txt:22]: I believe those two files should be be returned by the TF-M build, or exists as TF-M build properties that can be fetched when needed.

This is a bit to deep knowledge of the TF-M build structure.

File: boards/arm/nrf5340dk_nrf5340/CMakeLists.txt [hunk 3]
```diff
@@ -7,3 +7,73 @@ if (((CONFIG_BOARD_NRF5340PDK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340PDK_NRF5340_
...
+	# the image manifest
+	if(NOT DEFINED TFM_PUBLIC_KEY_FORMAT)
+		set(TFM_PUBLIC_KEY_FORMAT "full")
+	endif()
+
+	# Set srec_cat binary name
+	find_program(SREC_CAT srec_cat)
+	if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
+	    message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
+	endif()
```
Comment (tejlmand) [inline boards/arm/nrf5340dk_nrf5340/CMakeLists.txt:35]: I don't see `srec_cat` being used here ?

File: boards/arm/nrf9160dk_nrf9160/CMakeLists.txt [hunk 4]
```diff
@@ -5,3 +5,73 @@ if(CONFIG_BOARD_NRF52840_GPIO_RESET)
...
+	# the image manifest
+	if(NOT DEFINED TFM_PUBLIC_KEY_FORMAT)
+		set(TFM_PUBLIC_KEY_FORMAT "full")
+	endif()
+
+	# Set srec_cat binary name
+	find_program(SREC_CAT srec_cat)
+	if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
+	    message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
+	endif()
```
Comment (tejlmand) [inline boards/arm/nrf9160dk_nrf9160/CMakeLists.txt:33]: I don't see `srec_cat` being used here.

File: samples/tfm_integration/psa_level_1/boards/nrf5340pdk_nrf5340_cpuappns.overlay [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * TF-M.
+ */
+/delete-node/ &sram0_ns;
+/ {
+	reserved-memory {
+		sram0_ns: image_ns@20040000 {
+			reg = <0x20040000 DT_SIZE_K(192)>;
+		};
+	};
+};
```
Comment (tejlmand) [inline samples/tfm_integration/psa_level_1/boards/nrf5340pdk_nrf5340_cpuappns.overlay:32]: seems to lack a newline.

File: samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * TF-M.
+ */
+/delete-node/ &sram0_ns;
+/ {
+	reserved-memory {
+		sram0_ns: image_ns@20040000 {
+			reg = <0x20040000 DT_SIZE_K(192)>;
+		};
+	};
+};
```
Comment (tejlmand) [inline samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay:32]: seems to lack a newline.

**Review by tejlmand — CHANGES_REQUESTED** [review]
General impression is ok.
I see some repetition like everyone having to set `TFM_TARGET_PLATFORM` (which are almost aligned with Zephyr board folder name) and `TFM_PUBLIC_KEY_FORMAT.
And there are probably more.
Should we consider (later) to have a kind of Zephyr board -> TFM settings table, instead of each having to write the same CMake code ?

Also I think we need to look into a more proper handling of TF-M output files, but such cleanup can be done after this PR.

File: boards/arm/mps2_an521/CMakeLists.txt (hunk 2 above)
Comment (tejlmand) [reply boards/arm/mps2_an521/CMakeLists.txt:22]: can of course be improved later, just wanted to mention it.

**Comment by ioannisg** [comment]
@carlescufi @anangl @oyvindronningstad could you take a look at [this ](https://github.com/zephyrproject-rtos/zephyr/pull/29027/commits/7f80039772ea260a6503d7d33b88c99b8557568b) commit (docs)

File: samples/tfm_integration/tfm_ipc/boards/nrf9160dk_nrf9160ns.overlay [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ * used in this sample.
+ */
+
+/* Increase the size of the Secure Firmware (TF-M).
+ * This modification is not required at the moment,
+ * since TF-M region definitions are configured
+ * statically in the TF-M project.
+ */
+&sram0_s {
+	reg = <0x20000000 DT_SIZE_K(192)>;
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/boards/nrf9160dk_nrf9160ns.overlay:18]: Agreed, although this does not really play any role, at the moment, right? Since tFM has it's own build process that does not take DTS into account....

File: boards/arm/nrf5340dk_nrf5340/doc/index.rst [hunk 8]
```diff
@@ -200,6 +209,33 @@ The process requires the following steps:
...
+
+1. Build the Non-Secure Zephyr application
+   for the Application MCU using ``-DBOARD=nrf5340pdk_nrf5340_cpuappns`` and
+   ``CONFIG_BUILD_WITH_TFM=y`` in the application project configuration file.
+   The Zephyr build system will perform the following steps automatically:
+
+      * Build the Non-Secure firmware image as a regular Zephyr application
+      * Build a TF-M (secure) firmware image
+      * Optionally build a bootloader image (MCUboot)
+      * Merge the output binaries together
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/doc/index.rst:227]: I'm not sure if it will be clear that only the two firmware images are merged together and the bootloader one is not. Maybe it would be better to change the order of the last two points?

Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/doc/index.rst:227]: Yes! that's good point 👍

Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/doc/index.rst:226]: done, and also for nrf9160

File: boards/arm/nrf5340dk_nrf5340/CMakeLists.txt [hunk 9]
```diff
@@ -7,3 +7,83 @@ if (((CONFIG_BOARD_NRF5340PDK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340PDK_NRF5340_
...
+		set(TFM_IMAGE_VERSION_S 0.0.0+0)
+	endif()
+
+	if (NOT DEFINED TFM_IMAGE_VERSION_NS)
+		set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+	endif()
+
+	set(PREPROCESSED_FILE_S "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o")
+	set(PREPROCESSED_FILE_NS "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o")
+	set(TFM_MCUBOOT_DIR "${ZEPHYR_BASE}/../modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot")
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/CMakeLists.txt]: sure, done (for all platforms)

File: samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+/* Decrease the size of the Non-Secure Firmware (Zephyr),
+ * and move its starting address to the offset expected by
+ * TF-M.
+ */
+/delete-node/ &sram0_ns;
+/ {
+	reserved-memory {
+		sram0_ns: image_ns@20040000 {
+			reg = <0x20040000 DT_SIZE_K(192)>;
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay]: done

File: samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+ * used in this sample.
+ */
+
+/* Increase the size of the Secure Firmware (TF-M).
+ * This modification is not required at the moment,
+ * since TF-M region definitions are configured
+ * statically in the TF-M project.
+ */
+&sram0_s {
+	reg = <0x20000000 DT_SIZE_K(256)>;
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay]: done

File: samples/tfm_integration/tfm_ipc/boards/nrf9160dk_nrf9160ns.overlay (hunk 7 above)
[the remark this answers is not in the document]
Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/boards/nrf9160dk_nrf9160ns.overlay]: done

File: samples/tfm_integration/psa_level_1/sample.yaml [hunk 12]
```diff
@@ -5,7 +5,8 @@ sample:
 tests:
     sample.tfm_ipc:
         tags: introduction
-        platform_allow: mps2_an521_nonsecure lpcxpresso55s69_ns
+        platform_allow: mps2_an521_nonsecure lpcxpresso55s69_ns nrf5340pdk_nrf5340cpuappns
```
Comment (anangl) [inline samples/tfm_integration/psa_level_1/sample.yaml:8]:
```suggestion
        platform_allow: mps2_an521_nonsecure lpcxpresso55s69_ns nrf5340pdk_nrf5340_cpuappns
```

File: samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay [hunk 13]
```diff
@@ -28,18 +31,18 @@
...
 			label = "image-0";
-			reg = <0x00011000 0x00040000>;
+			reg = <0x00013000 0x00038000>;
 		};
 		/* Non-secure image primary slot */
-		slot1_partition: partition@00051000 {
+		slot1_partition: partition@0004B000 {
 			label = "image-1";
-			reg = <0x00051000 0x0002E000>;
+			reg = <0x0004B000 0x0002A000>;
```
Comment (erwango) [inline samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay:45]: Is that required ?
If yes, I think I'd be better moved in dedicated commit anyway

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt [hunk 14]
```diff
@@ -16,49 +16,50 @@ if (CONFIG_BUILD_WITH_TFM)
...
 			 --align 1
 			 -v ${TFM_IMAGE_VERSION_S}
+			 --pad
+			 --pad-header
 			 ${ADD_NS_IMAGE_MIN_VER}
-			 ${ADD_SECURITY_COUNTER_S}
+			 -s auto
 			 -H 0x400
-			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/STM_NUCLEO_L552ZE_Q/tfm_s.bin
+			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/STM/NUCLEO_L552ZE_Q/tfm_s.bin
```
Comment (erwango) [inline boards/arm/nucleo_l552ze_q/CMakeLists.txt:48]: Should is be lower case here to match TFM_TARGET_PLATFORM value ?

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt [hunk 15]
```diff
@@ -60,6 +66,6 @@ if (CONFIG_BUILD_WITH_TFM)
 		COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/tfm/bin/bl2.bin ${CMAKE_BINARY_DIR}/mcuboot.bin
 
 		#Execute post build script postbuild.sh
-		COMMAND ${CMAKE_BINARY_DIR}/tfm/postbuild.sh
+		COMMAND ${CMAKE_BINARY_DIR}/tfm/postbuild.sh ${COMPILER_FULL_PATH}
```
Comment (erwango) [inline boards/arm/nucleo_l552ze_q/CMakeLists.txt:69]: This is a nice improvement.
Please also update the following note: https://github.com/zephyrproject-rtos/zephyr/blob/3ce7afdaaa005dc0f46d721ad77d0c275ce0df9f/samples/tfm_integration/tfm_ipc/README.rst#L132

**Review by erwango — CHANGES_REQUESTED** [review]
Few comments

File: boards/arm/mps2_an521/CMakeLists.txt [hunk 16]
```diff
@@ -35,43 +36,43 @@ if (CONFIG_BUILD_WITH_TFM)
...
-			 --layout ${PREPROCESSED_FILE}_ns.c
+		COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/wrapper/wrapper.py
+			 --layout ${PREPROCESSED_FILE_NS}
 			 -k ${CONFIG_TFM_KEY_FILE_NS}
 			 --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
 			 --align 1
 			 -v ${TFM_IMAGE_VERSION_NS}
+			 -s auto
 			 ${ADD_S_IMAGE_MIN_VER}
 			 ${ADD_SECURITY_COUNTER_NS}
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply boards/arm/mps2_an521/CMakeLists.txt:62]: yeap, just that I missed removing it. It does not affect the code, but it is misleading.
Removed now.

Actually, we should be allowing the user to set this e.g. via Kconfig and pass this to both the tf-m build and/or the post-build signing utility....

File: samples/tfm_integration/psa_level_1/sample.yaml (hunk 12 above)
Comment (ioannisg) [reply samples/tfm_integration/psa_level_1/sample.yaml:8]: good catch, fixed now, thanks.

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt (hunk 14 above)
Comment (ioannisg) [reply boards/arm/nucleo_l552ze_q/CMakeLists.txt:48]: Nope, this should be compliant with upstream TFM - this is how it looks like in the new build system.

File: samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay (hunk 13 above)
Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay:45]: There is already a dedicated commit, @erwango concerning this, in the trusted-firmware-m repository.
https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/15/commits/8dca035c06a23907998c2c550f55fdc68136bc86

Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay:45]: So this is required, because the TFM layout is changed now (or, strictly speaking , it will be changed if the upstream fixes get merged). I can move this to a separate commit, no problem.

Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay:45]: moved, now.

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt (hunk 15 above)
Comment (ioannisg) [reply boards/arm/nucleo_l552ze_q/CMakeLists.txt:68]: done :)

File: boards/arm/nucleo_l552ze_q/CMakeLists.txt (hunk 14 above)
Comment (erwango) [reply boards/arm/nucleo_l552ze_q/CMakeLists.txt:48]: ok, txs

File: .buildkite/run.sh [hunk 17]
```diff
@@ -53,6 +53,10 @@ ccache -s
 # lpcxpresso11u68 boards
 pip3 install lpc_checksum
 
+# Temporary fix: Install imgtool, needed for MCUboot to sign images
+# when builindg the TF-M integration samples
+pip3 install imgtool
```
Comment (mbolivar-nordic) [inline .buildkite/run.sh:58]: Can't you put this in one of the pip install lines in .github/foo.yaml instead?

File: scripts/requirements-build-test.txt [hunk 18]
```diff
@@ -22,3 +22,6 @@ mypy
 
 # used for mocking functions in pytest
 mock>=4.0.1
+
+# used in MCUboot for signing and key management
+imgtool>=1.6.0
```
Comment (mbolivar-nordic) [inline scripts/requirements-build-test.txt:27]: Not sure if the answer is yes or no, but can you check if cmake/mcuboot.cmake support for finding imgtool via PATH or from mcuboot/scripts should be extracted instead of making this a generic requirement?

Comment (ioannisg) [reply scripts/requirements-build-test.txt:27]: The problem is that MCUboot in this case, @mbolivar-nordic , is built via TF-M, which is an external project. It is not build with Zephyr.

Comment (ioannisg) [reply scripts/requirements-build-test.txt:27]: ...in other words,  cmake/mcuboot.cmake, is not processed at all.

File: .buildkite/run.sh (hunk 17 above)
Comment (ioannisg) [reply .buildkite/run.sh:58]: In that case, I suppose it should be in `./github/workflows/sanitycheck_tests.yml`, but I am not sure if this would be better that having this temporary hack in .buildkite (which could be eventually removed, once we have a new image for CI).
Could you elaborate?

@galak any ideas (since you initially suggested this)?

File: scripts/requirements-build-test.txt (hunk 18 above)
Comment (mbolivar-nordic) [reply scripts/requirements-build-test.txt:27]: > ...in other words, cmake/mcuboot.cmake, is not processed at all.

Yes, of course, I'm asking if factoring out that code to find imgtool into shared code would be a better approach. That is why I said "extracted".

File: .buildkite/run.sh (hunk 17 above)
Comment (mbolivar-nordic) [reply .buildkite/run.sh:58]: > Could you elaborate?

We have a bunch of different pip install lines throughout the github workflows which is already confusing for people and causing problems.

For example, I had to send https://github.com/zephyrproject-rtos/zephyr/pull/28922 because https://github.com/zephyrproject-rtos/zephyr/pull/28047 passed CI by editing one pip install line in one workflow used by CI, but the docs daily build used a different environment. I don't like the situation already, it's messy and people don't know where to put pip install lines anymore as seen from this example, and adding yet another place where we have to remember they might be found is not making things better.

File: scripts/requirements-build-test.txt (hunk 18 above)
Comment (mbolivar-nordic) [reply scripts/requirements-build-test.txt:27]: By putting this here it looks like you are basically forcing everybody who uses zephyr+mcuboot+signing via the build system to get imgtool from pip instead of from the mcuboot repository. When I added mcuboot.cmake, it was a deliberate choice to favor imgtool installed via pip over the one found in the mcuboot repository because imgtool was **not** installed by default in the zephyr requirements.txt. This is changing that default.

File: .buildkite/run.sh (hunk 17 above)
Comment (ioannisg) [reply .buildkite/run.sh:58]: FIne, I will try with adding this, then, in ./github/workflows/sanitycheck_tests.yml if this is OK with you.

File: scripts/requirements-build-test.txt (hunk 18 above)
Comment (ioannisg) [reply scripts/requirements-build-test.txt:27]: OK, so to conclude, what would be your suggestion @mbolivar-nordic ?

I cannot, really, hack the external project to look for imgtool from anywhere else than PIP; that will be a nightmare to maintain. I am OK with removing it from requirements.txt, since I already added a doc entry in the TFM samples regarding the python requirements for building TFM binaries (FYI @microbuilder ) so maybe we can afford to skip adding this here...

No strong opinion, but I would also like @microbuilder to comment.

File: .buildkite/run.sh (hunk 17 above)
Comment (mbolivar-nordic) [reply .buildkite/run.sh:58]: > FIne, I will try with adding this, then, in ./github/workflows/sanitycheck_tests.yml if this is OK with you.

Sounds good, at least it doesn't add a new place to find pip install lines. Thanks.

File: scripts/requirements-build-test.txt (hunk 18 above)
Comment (mbolivar-nordic) [reply scripts/requirements-build-test.txt:27]: > OK, so to conclude, what would be your suggestion @mbolivar-nordic ?

My suggestion is to find imgtool in host-tools.cmake by extracting the code in mcuboot.cmake and setting an IMGTOOL CMake variable to something you can run to run imgtool.

My point is that if imgtool is a host tool that is needed in multiple places now (as this PR seems to be making the case), it should probably be treated like any other host tool in the sense of its discovery getting moved to host-tools.cmake. Previously I didn't do this because mcuboot.cmake was a special purpose use case.

cc @tejlmand

File: .buildkite/run.sh (hunk 17 above)
Comment (ioannisg) [reply .buildkite/run.sh]: done now, @mbolivar-nordic , hope this does not break CI

Comment (ioannisg) [reply .buildkite/run.sh]: Turns out, this won't work, @mbolivar-nordic, as this workflow activates only upon changes directly on sanity check scripts. I don't think it makes sense to add a new workflow here - perhaps better to revert to the initial workaround (patching .buildkite), as there is already a similar workaround there (for `lpc_checksum`).
I can file an enhancement issue to track the removal of this workaround.

**Review by anangl — APPROVED** [review]

**Comment by ioannisg** [comment]
DNM until the manifest pointer is updated.

File: samples/tfm_integration/psa_level_1/sample.yaml [hunk 19]
```diff
@@ -5,7 +5,8 @@ sample:
 tests:
     sample.tfm_ipc:
```
Comment (erwango) [inline samples/tfm_integration/psa_level_1/sample.yaml:6]: Not related to current change, but might deserve a fix

**Comment by microbuilder** [comment]
@ioannisg Concerning the imgtool dependency, I'm OK to only have this requirement documented as mandatory for TF-M, since there are other requirements to meet anyway, rather than forcing the install on everyone for this edge case.

**Comment by ioannisg** [comment]
> @ioannisg Concerning the imgtool dependency, I'm OK to only have this requirement documented as mandatory for TF-M, since there are other requirements to meet anyway, rather than forcing the install on everyone for this edge case.

Thanks. Then I'll remove the edit in the requirements_.txt

File: samples/tfm_integration/psa_level_1/sample.yaml (hunk 19 above)
Comment (ioannisg) [reply samples/tfm_integration/psa_level_1/sample.yaml:6]: Fixing that too, thanks.

File: scripts/requirements-build-test.txt (hunk 18 above)
Comment (ioannisg) [reply scripts/requirements-build-test.txt:27]: @mbolivar-nordic based on @microbuilder 's comment below, we 'll remove this edit in the requirements-build-test.txt.

**Review by erwango — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Comment by ioannisg** [comment]
@tejlmand please, take a look so this review could be finalized.

File: .buildkite/run.sh (hunk 17 above)
Comment (mbolivar-nordic) [reply .buildkite/run.sh:58]: >  I can file an enhancement issue to track the removal of this workaround.

Please do, thanks.

File: scripts/requirements-build-test.txt (hunk 18 above)
Comment (mbolivar-nordic) [reply scripts/requirements-build-test.txt]: > we 'll remove this edit in the requirements-build-test.txt.

Thanks.

**Review by mbolivar-nordic — APPROVED** [review]
My feedback is addressed.

**Comment by ioannisg** [comment]
> My feedback is addressed.

Yeap, thanks.

File: samples/tfm_integration/psa_level_1/README.rst [hunk 20]
```diff
@@ -140,13 +140,46 @@ and non-secure (``zephyr.hex``) images wth a J-Link as follows:
...
+(``-DBOARD=nrf5340pdk_nrf5340_cpuappns`` or ``-DBOARD=nrf9160dk_nrf9160ns``).
+
+   Example, for nRF9160, using ``cmake`` and ``ninja``
+
+   .. code-block:: bash
+
+      cd <ZEPHYR_ROOT>/samples/tfm_integration/psa_level_1/
+      rm -rf build
+      mkdir build && cd build
+      cmake -GNinja -DBOARD=nrf9160dk_nrf9160ns ..
```
Comment (tejlmand) [inline samples/tfm_integration/psa_level_1/README.rst:163]: Just wondering, why not just do:
```suggestion
      cmake -GNinja -DBOARD=nrf9160dk_nrf9160ns -Bbuild .
```
that will create the `build` folder and thus save the `mkdir builld && cd build` step

Note, this is not blocking in this PR, as I know this description has been used elsewhere.

File: samples/tfm_integration/tfm_ipc/CMakeLists.txt [hunk 21]
```diff
@@ -14,20 +14,19 @@ elseif (CONFIG_TFM_BL2_FALSE)
 endif()
 
 # Add "tfm" as an external project via the TF-M module's cmake file
-if(${TFM_TARGET_PLATFORM} STREQUAL "STM_NUCLEO_L552ZE_Q")
+if(${TFM_TARGET_PLATFORM} STREQUAL "stm/nucleo_l552ze_q")
```
Comment (tejlmand) [inline samples/tfm_integration/tfm_ipc/CMakeLists.txt:17]: Just noticing, using `${TFM_TARGET_PLATFORM}` without qoutes is dangerous in case `TFM_TARGET_PLATFORM` has not been defined.
Will give an error like:
```
CMake Error at samples/tfm_integration/tfm_ipc/CMakeLists.txt:17 (if):
  if given arguments:

    "STREQUAL" "stm/nucleo_l552ze_q"

  Unknown arguments specified
```
whereas qouting would be comparing an empty string, which would be a valid comparison.
```suggestion
if("${TFM_TARGET_PLATFORM}" STREQUAL "stm/nucleo_l552ze_q")
```
and of cause be false, if `TFM_TARGET_PLATFORM` os empty/undefined.

Note, as this is old code (only changing right side string), then it has probably not been a problem up til noew and can thus be cleaned up later.

**Review by tejlmand — APPROVED** [review]
Changed approved.

I still think the TFM integration needs an overhaul / cleanup in general.
It was rushed in for 2.3, and haven't gotten the required cleanup afterwards, but thas is outside the scope of this PR.

File: samples/tfm_integration/psa_level_1/README.rst (hunk 20 above)
Comment (ioannisg) [reply samples/tfm_integration/psa_level_1/README.rst:163]: Yeap, we sort of use the old description in most of the cases, so while this is a good suggestion, I'd prefer not to implement it for this particular sample (but we could apply tree-wide later)

File: samples/tfm_integration/tfm_ipc/CMakeLists.txt (hunk 21 above)
Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/CMakeLists.txt:17]: Thanks @tejlmand , this is good feedback. I can actually fix this in the current PR.

File: samples/tfm_integration/psa_level_1/README.rst (hunk 20 above)
Comment (tejlmand) [reply samples/tfm_integration/psa_level_1/README.rst:163]: yeap. Fully accepted, just wanted to point it out.

File: samples/tfm_integration/tfm_ipc/CMakeLists.txt (hunk 21 above)
Comment (ioannisg) [reply samples/tfm_integration/tfm_ipc/CMakeLists.txt]: Fixed now.
